java/spark/src/main/java/com/logicalclocks/hsfs/spark/util/StorageConnectorUtils.java

```diff
@@ -66,7 +66,12 @@ public Dataset<Row> read(StorageConnector.HopsFsConnector connector,
   public Dataset<Row> read(StorageConnector.S3Connector connector, String dataFormat,
                            Map<String, String> options, String path) throws FeatureStoreException, IOException {
     connector.update();
-    return SparkEngine.getInstance().read(connector, dataFormat, options, path);
+    Map<String, String> readOptions = connector.sparkOptions();
+    // merge user spark options on top of default spark options
+    if (options != null && !options.isEmpty()) {
+      readOptions.putAll(options);
+    }
+    return SparkEngine.getInstance().read(connector, dataFormat, readOptions, path);
```

Comment:
I don't think this works for the `fs.s3a.endpoint` property. I think for these set of properties we need to set them in the hadoop configuration here: https://github.com/logicalclocks/feature-store-api/blob/3d90644a5341046695732c0687c1dcb6e758a5a0/java/spark/src/main/java/com/logicalclocks/hsfs/spark/engine/SparkEngine.java#L727

We need to check, but I believe we can pass the options both here and on the method I linked above.